### ISSUE TITLE
cant get doc_type to apply

### ISSUE THREAD

**Issue by mikedevita** [pcfens/puppet-filebeat#60]
using foreman for my ENC with data defined as such, i cant get the doc_type to apply.

```bash
root@phxlv-log01:/usr/share/filebeat/bin# ./filebeat --version
filebeat version 5.0.1 (amd64), libbeat 5.0.1
```

**filebeat::outputs (hash)**
```yaml
logstash:
  index: filebeat
  hosts:
  - phxlv-log01.vm.slan.devita.co:5044
```
**filebeat::prospectors (hash)**
```yaml
syslogs:
  doc_type: syslog
  paths:
  - "/var/log/syslog"
authlogs:
  doc_type: authlog
  paths:
  - "/var/log/auth.log"
```

the file in `/etc/filebeats/conf.d/authlogs.yml` looks like this

```yaml
---
filebeat:
  prospectors:
    - input_type: log
      paths:
        - /var/log/auth.log
      encoding: plain
      fields_under_root: false
      scan_frequency: 10s
      harvester_buffer_size: 16384
      max_bytes: 10485760
```

**Comment by pcfens** [comment]
I'm not able to re-create the issue in hiera (what I've done is below), and I'm not a foreman user.

The manifest is only `include ::filebeat`.

The single matching file in hiera is:
```yaml
---
filebeat::outputs:
  logstash:
    index: filebeat
    hosts:
      - logstash.example.com:5044

filebeat::prospectors:
  syslogs:
    doc_type: syslog
    paths:
      - "/var/log/syslog"
  authlogs:
    doc_type: authlog
    paths:
      - "/var/log/auth.log"
```

The resulting authlogs.yml begins with
```yaml
---
filebeat:
  prospectors:
    - input_type: log
      paths:
        - /var/log/auth.log
      encoding: plain
      fields_under_root: false
      document_type: authlog
      scan_frequency: 10s
      harvester_buffer_size: 16384
      max_bytes: 10485760
```
Any chance this gives you any ideas of things I can look in to? If not, have you run across a similar issue with other modules that they were able to fix? We might be able to see what they did to fix it.

**Comment by mikedevita** [comment]
yeah hmm im not sure thats basically what i have in foreman as a hash..

**Comment by mikedevita** [comment]
i got it resolved, upgrading to 0.8.6 fixed it (I was on 0.8.1) woops..
